### PR TITLE
Memory & Resync Fixes

### PR DESCRIPTION
**Pull request by mplorentz**
This is a conglomeration of fixes @boreq and I have been making to the migration branch this week. In particular it:
- #459 Lowers the memory usage to acceptable levels. This was done by adding some new options to go-ssb to limit the number of feeds we replicate concurrently, and by limited the number of messages we fetch from a given feed to 1000 at a time.
- Fixes the majority of cases where replication would halt while the app was running (#524). This allows the migration to run smoothly without the user needing to restart the app.
- Makes the migration screen show again after an app restart if the app was closed while it was running.
- Tweaks the did_bot_stats analytics event so that it fires every time statistics are computed.
- Adds a new `written_at` column to the messages table in SQLite. This new column is used to compute the number of messages that have been written to SQLite recently instead of `received_at`, because sometimes `received_at` and `written_at` can be drastically different times and it was causing the Connected Peers List in the sidebar to show 0 new messages during a resync.
- Fixes timestamp parsing in go-ssb to fix some feeds being shown in the future. #520

### CODE REVIEW

File: GoSSB/Sources/api-ios.go [hunk 1]
```diff
@@ -315,6 +318,8 @@ func ssbBotInit(config string, notifyBlobReceivedFn uintptr, notifyNewBearerToke
 		}),
 		mksbot.DisableEBT(true),
 		mksbot.WithPublicAuthorizer(newAcceptAllAuthorizer()),
+		mksbot.WithNumberOfConcurrentReplicationsPerPeer(25),
```
Comment (boreq) [inline GoSSB/Sources/api-ios.go:321]: I see that we went down to 25, I think that is reasonable.

**Review by boreq — APPROVED** [review]
Looks good on the Go side of things, go-ssb version is correct.

File: GoSSB/Sources/api-ios.go (hunk 1 above)
Comment (mplorentz) [reply GoSSB/Sources/api-ios.go:321]: Yeah given that we are currently bottlenecked by SQLite I am trying to make replication a little slower which gives SQLite a few more resources. I may tweak it again if I can get significantly more performance out of SQLite.

Btw I think these options would be great to contribute back to cryptoscope/ssb in some way if it makes sense. It gives anyone using this as a library some coarse control over memory usage which seems important on any device that isn't a modern desktop.

**Comment by mplorentz** [comment]
I found a case where the migration could get run again if we changed the bot version in the future, so I fixed that and added it to this PR.

**Comment by boreq** [comment]
> I found a case where the migration could get run again if we changed the bot version in the future, so I fixed that and added it to this PR.

I actually experienced this today when running my new branch where I changed the version. Good to see it fixed.

**Comment by mplorentz** [comment]
I'm going to go ahead and merge this since it's been open for a week.

**Pull request merged**